src/composables/fungibleTokens.ts

```diff
@@ -0,0 +1,491 @@
+   *  - convertedBalance: The balance of the token.
+   *  - symbol: The symbol of the token (i.e. "AE").
+   *  - balanceCurrency: The balance currency of the token.
+   *  - contractId: The contract ID of the Aeternity token contract.
+   */
+  function getAeternityToken({
+    balanceCurrency,
+    tokenBalance,
+  }: { balanceCurrency: BigNumber | number, tokenBalance: BigNumber | Balance }): IToken | IAsset {
+    const aePublicDataExists = aePublicData.value && Object.keys(aePublicData.value).length > 0;
+
+    return {
+      ...(aePublicDataExists ? aePublicData : {} as IAsset),
+      convertedBalance: tokenBalance,
+      symbol: AETERNITY_SYMBOL,
+      balanceCurrency,
+      contractId: AETERNITY_CONTRACT_ID,
+    } as IToken | IAsset;
+  }
+
+  /**
+   * Loads available tokens from current active network.
+   */
+  async function loadAvailableTokens(): Promise<void> {
+    try {
+      await watchUntilTruthy(() => middleware.value);
+      const response: IToken[] = camelcaseKeysDeep(await fetchJson(
+        `${activeNetwork.value.middlewareUrl}/aex9/by_name`,
+      ));
+
+      if (isEmpty(response) || typeof response !== 'object') {
+        availableTokens.value = {};
+      } else {
+        const fetchedTokens: ITokenList = response.reduce(
+          (obj, token: IToken) => ({
+            ...obj, [token.contractId]: token,
+          }), {},
+        );
+
+        availableTokens.value = fetchedTokens;
+      }
+    } catch (error) {
+      handleUnknownError(error);
+    }
+  }
+
+  /**
+   * Loads the address token balances.
+   */
+  async function loadTokenBalances(address: string) {
+    activeAddress.value = address;
+    try {
+      await watchUntilTruthy(() => middleware.value);
+      if (isEmpty(availableTokens.value)) {
+        await loadAvailableTokens();
+      }
+
+      // TODO: switch to mdw v2 route, once sync issue is resolved
+      // const _tokens = await getAllPages(
+      //   () => fetchJson(
+      //     `${activeNetwork.value.middlewareUrl}/v2/aex9/account-balances/${address}?limit=100`,
+      //   ),
+      //   middleware.value.fetchByPath,
+      // );
+
+      const _tokens: any[] = await fetchJson(
+        `${activeNetwork.value.middlewareUrl}/aex9/balances/account/${address}`,
+      );
+
+      if (isEmpty(_tokens) || typeof _tokens !== 'object') {
+        return;
+      }
+
+      const balances = _tokens.map(({ amount, contract_id: contractId }) => {
+        const token = availableTokens.value[contractId];
+        if (!token) return null;
+        const balance = convertToken(amount, -token.decimals);
+        const convertedBalance = balance.toFixed(2);
+        const objectStructure = {
+          ...token,
+          value: contractId,
+          text: `${convertedBalance} ${token.symbol}`,
+          contractId,
+          balance,
+          convertedBalance,
+        };
+
+        return objectStructure as any;
```

Comment:
I know that I'm annoying, but as always: I suggest to avoid using `as any`.
Also the object can be returned directly without creating `objectStructure` constant.